File: tools_box/tools_box/report/customer_commission_report/customer_commission_report.py
# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):
	if not filters: filters ={}
	columns = ["Item:Link/Item:300","Invoice Date:Date:110","Sales Invoice::100","Customer:Link/Customer:200","Qty:Float:50","Amount:Currency:100"]
	conditions = ""
	if filters.get("from_date"):
		conditions += " AND si.posting_date >= %(from_date)s"
	if filters.get("to_date"):
		conditions += " AND si.posting_date <= %(to_date)s"
	if filters.get("customer"):
		conditions += " AND si.customer = %(customer)s"
	if filters.get("item"):
		conditions += " AND sii.item_code = %(item)s"
	data = frappe.db.sql("""SELECT CONCAT(sii.item_name,' (',sii.item_code,')'),si.posting_date,si.name,si.customer,sii.qty,sii.amount FROM `tabSales Invoice` si, `tabSales Invoice Item` sii WHERE si.name = sii.parent {0} ORDER BY sii.item_name,si.posting_date""".format(conditions),filters)
	return columns, data

File: tools_box/tools_box/report/customer_commission_report/customer_commission_report.js
// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt

frappe.query_reports["Customer Commission Report"] = {
	"filters": [
        {
                        fieldname: "from_date",
                        label: __("From Date"),
                        fieldtype: "Date",
                },
        {
                        fieldname: "to_date",
                        label: __("To Date"),
                        fieldtype: "Date",
                },
        {
                        fieldname: "item",
                        label: __("Item"),
                        fieldtype: "Link",
                        options: "Item",
                },
        {
                        fieldname: "customer",
                        label: __("Customer"),
                        fieldtype: "Link",
                        options: "Customer",
                },
        {
                        fieldname: "commission_rate",
                        label: __("Commission Rate"),
                        fieldtype: "Float",
			default: 3,
                },
	]
}
